### DISCUSSION TITLE
Two-Column Page With Automatic Overflow

### DISCUSSION THREAD

**Discussion by DCarlson12** [QuestPDF/QuestPDF#413] [Q&A] [answered]
Hi, I'm trying to figure out how to create a document page that shows 2 columns per page. I found a discussion [here](https://github.com/QuestPDF/QuestPDF/discussions/21) about something similar, but it had a pre-determined amount of items to split between the columns and then break pages. It looks like [this ](https://github.com/QuestPDF/QuestPDF/discussions/60)discussion has asked for this same layout handling.

I am trying to make a menu generator that will have named sections and will automatically overflow to the second column once the first one is out of space instead of flowing onto the next page. Once the second column on the page runs out of space, it should automatically flow to the left hand column on the next page. The menu sections don't need to have ensured space, the items under any section may flow into the next column or page.

Here is an example in Word I mocked up using the Layout > Two Column option. This is the exact behavior I am after:

![image](https://user-images.githubusercontent.com/64709964/201770848-cd6f6d3e-b056-404d-b07a-32b2d77e3813.png)

I cannot for the life of me figure out how I would make this work. The number of menu sections and items in a section will vary and can not be know until the model information is passed to the document to compose it. Can anyone provide some pointers on how this could be done? Thank you.

EDIT: Okay it looks like a lot of people are asking for this behavior. Probably wasn't using the best search terms in the discussions and issues.

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

The functionality you described is now available through the [MultiColumn](https://www.questpdf.com/api-reference/multi-column.html) element, introduced in version 2024.7.0. Content automatically flows from the first column to the next, and then onto the following page — exactly the behavior you were after.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
